src/components/links/LinkBase.tsx

```diff
@@ -0,0 +1,152 @@
+export interface LinkProps extends Pick<RouterLinkProps, RouterLinkPropsRequired> {
+  children: React.ReactNode;
+  className?: string;
+  shouldBlurAfterClick?: boolean;
+  isDisabled?: boolean;
+  icon?: React.ReactNode;
+  isExternal?: boolean;
+  onClick?: (event: React.MouseEvent<HTMLAnchorElement>) => void;
+  shouldPreventDefault?: boolean;
+  hasExternalIcon?: boolean;
+  shouldStopPropagation?: boolean;
+  target?: HTMLAttributeAnchorTarget;
+}
+
+function LinkBaseWithRef(props: LinkProps, ref: React.ForwardedRef<HTMLAnchorElement>) {
+  const {
+    children,
+    shouldBlurAfterClick = false,
+    isDisabled = false,
+    icon,
+    isExternal: isExternalProp = false,
+    onClick,
+    shouldPreventDefault = false,
+    hasExternalIcon = !icon,
+    shouldStopPropagation = false,
+    to,
+    ...rest
+  } = props;
+
+  const toAsString =
+    typeof to === 'string' ? to : `${to.pathname ?? ''}${to.search ?? ''}${to.hash ?? ''}`;
+
+  const isExternal = isExternalProp || toAsString.startsWith('http');
+
+  const handleClick = React.useCallback(
+    (event: React.MouseEvent<HTMLAnchorElement>) => {
+      if (shouldBlurAfterClick) {
+        event.currentTarget.blur();
+      }
+
+      if (shouldPreventDefault || isDisabled) {
+        event.preventDefault();
+      }
+
+      if (shouldStopPropagation) {
+        event.stopPropagation();
+      }
+
+      if (onClick && !isDisabled) {
+        onClick(event);
+      }
+    },
+    [onClick, shouldBlurAfterClick, shouldPreventDefault, shouldStopPropagation, isDisabled],
+  );
+
+  if (isExternal) {
+    return (
+      <a
+        rel="noopener noreferrer"
+        target="_blank"
+        {...rest}
+        href={toAsString}
+        onClick={handleClick}
+        ref={ref}>
+        {icon}
+        {children}
+        {hasExternalIcon && <ExternalIcon className="sw-ml-1" />}
+      </a>
+    );
+  }
+
+  return (
+    <RouterLink ref={ref} {...rest} onClick={handleClick} to={to}>
+      {icon}
+      {children}
+    </RouterLink>
+  );
+}
+
+const ExternalIcon = styled(OpenNewTabIcon)`
+  color: var(--echoes-color-background-accent-default);
+  margin-left: 4px;
+`;
+
+export const LinkBase = React.forwardRef(LinkBaseWithRef);
+
+export const LinkBaseStyled = styled(LinkBase)`
+  color: var(--color);
+  text-decoration-color: var(--color);
+
+  &:visited {
+    color: var(--color);
+  }
+
+  &:hover,
+  &:focus,
+  &:active {
+    color: var(--active);
+    text-decoration-color: 1px solid var(--active);
+
+    svg {
+      color: var(--active);
+    }
+  }
+
+  &:focus,
+  &:focus-visible {
+    background-color: var(--echoes-color-background-accent-weak);
+    outline: var(--echoes-color-border-focus) solid 2px;
+    outline-offset: var(--echoes-dimension-space-25);
+  }
+
+  & > svg {
+    vertical-align: text-bottom !important;
+  }
+
+  ${({ icon }) =>
+    icon &&
+    css`
+      & > svg,
+      & > img {
+        margin-right: 4px;
+      }
+    `};
+`;
```

Comment:
We can also add the @media print query from SC to ensure links don't have colors when printed:

```
@media print {
    &,
    &:visited,
    &:hover,
    &:focus,
    &:active {
      color: ${themeColor('currentColor')};
      text-decoration: none;
    }
  }

```

Reply:
Should the themecolor be replaced with 'currentColor'?

Reply:
yes :)